src/chocolatey.resources/helpers/functions/Install-ChocolateyInstallPackage.ps1

```diff
@@ -287,60 +293,77 @@ Pro / Business supports a single, ubiquitous install directory option.
-    } catch {
+    }
+    catch {
       Write-Warning "Unable to generate `'$ignoreFile`'"
     }
   }
 
   $workingDirectory = Get-Location -PSProvider "FileSystem"
   try {
     $workingDirectory = [System.IO.Path]::GetDirectoryName($fileFullPath)
-  } catch {
+  }
+  catch {
     Write-Warning "Unable to set the working directory for installer to location of '$fileFullPath'"
     $workingDirectory = $env:TEMP
   }
 
   try {
     # make sure any logging folder exists
     $pattern = "(?:['`"])([a-zA-Z]\:\\[^'`"]+)(?:[`"'])|([a-zA-Z]\:\\[\S]+)"
-    $silentArgs, $additionalInstallArgs | %{ Select-String $pattern -input $_ -AllMatches } |
-      % { $_.Matches } | % {
-        $argDirectory = $_.Groups[1]
-        if ($argDirectory -eq $null -or $argDirectory -eq '') { continue }
-        $argDirectory = [System.IO.Path]::GetFullPath([System.IO.Path]::GetDirectoryName($argDirectory))
-        Write-Debug "Ensuring '$argDirectory' exists"
-        if (![System.IO.Directory]::Exists($argDirectory)) { [System.IO.Directory]::CreateDirectory($argDirectory) | Out-Null }
-      }
-  } catch {
+    $silentArgs, $additionalInstallArgs | % { Select-String $pattern -input $_ -AllMatches } |
+    % { $_.Matches } | % {
+      $argDirectory = $_.Groups[1]
+      if ($argDirectory -eq $null -or $argDirectory -eq '') { continue }
+      $argDirectory = [System.IO.Path]::GetFullPath([System.IO.Path]::GetDirectoryName($argDirectory))
+      Write-Debug "Ensuring '$argDirectory' exists"
+      if (![System.IO.Directory]::Exists($argDirectory)) { [System.IO.Directory]::CreateDirectory($argDirectory) | Out-Null }
+    }
+  }
+  catch {
     Write-Debug "Error ensuring directories exist -  $($_.Exception.Message)"
   }
 
   if ($fileType -like 'msi') {
-    $msiArgs = "/i `"$fileFullPath`""
+    $msiArgs = if($overrideArguments){
+      Write-Host "Overriding package arguments with '$additonalInstallArgs' (replacing '$silentArgs')"
+      "$msiArgs $additionalInstallArgs"
+    } else {
+      "$msiArgs $silentArgs $additionalInstallArgs"
+    }
+
+    $env:ChocolateyExitCode = Start-ChocolateyProcessAsAdmin "$msiArgs" "$($env:SystemRoot)\System32\msiexec.exe" -validExitCodes $validExitCodes -workingDirectory $workingDirectory
+  }
+
+  if ($fileType -like 'msp') {
+    $msiArgs = '/update "{0}"' -f $fileFullPath
     if ($overrideArguments) {
       Write-Host "Overriding package arguments with '$additionalInstallArgs' (replacing '$silentArgs')";
       $msiArgs = "$msiArgs $additionalInstallArgs";
-    } else {
+    }
+    else {
       $msiArgs = "$msiArgs $silentArgs $additionalInstallArgs";
     }
 
-    $env:ChocolateyExitCode = Start-ChocolateyProcessAsAdmin "$msiArgs" "$($env:SystemRoot)\System32\msiexec.exe" -validExitCodes $validExitCodes -workingDirectory $workingDirectory
+    $env:ChocolateyExitCode = Start-ChocolateyProcessAsAdmin $msiArgs "$($env:SystemRoot)\System32\msiexec.exe" -validExitCodes $validExitCodes -workingDirectory $workingDirectory

```

Comment:
This change is interesting, as on line 335 above, the same thing hasn't been done, i.e. in one place we have `"$msiArgs"` and here we have `$msiArgs` so which one should we be using here?

Reply:
Quotes aren't necessary either way. My own preference is to avoid them unless they have a specific purpose, especially as they force a type conversion which isn't always appropriate. (Imagine calling `.ToString()` on everything you pass between methods or something, it's just kind of weird.)